### PR TITLE
feat: add deselected parameter in click event

### PR DESCRIPTION
**Pull request by m1ga**
**JIRA**: https://jira.appcelerator.org/browse/MOD-2545

**Description**:
Since the click event will fire twice when you click a different annotation I've added a parameter to see if the source was deselected or not. Same for clicking on the map instead of a second annotation.

```
var Map = require('ti.map');

var win = Ti.UI.createWindow({
	height: Ti.UI.FILL,
	width: Ti.UI.FILL
});
var map_view = Map.createView();


var south_korea = Map.createAnnotation({
	latitude: '36.792452',
	longitude: '127.995333',
	title: 'South Korea',
	pincolor: Map.ANNOTATION_BLUE
});

var north_korea = Map.createAnnotation({
	latitude: '39.763824',
	longitude: '126.856541',
	title: 'North Korea',
	pincolor: Map.ANNOTATION_RED
});

map_view.addAnnotation(south_korea);
map_view.addAnnotation(north_korea);

map_view.region = {
	latitude: '38.286344',
	longitude: '127.111459',
	zoom: 6
};

map_view.addEventListener('click', function(e) {
	Ti.API.info('Click ', e.title, ' clicksource: ', e.clicksource, ' type: ', e.type, " deselected: ", e.deselected);
});

win.add(map_view);

win.open();
```

* click first annoation
* click second annotation
* click map

output:
```
[INFO]  Click  North Korea  clicksource:  pin  type:  click  deselected:  false

[INFO]  Click  North Korea  clicksource:  pin  type:  click  deselected:  true
[INFO]  Click  South Korea  clicksource:  pin  type:  click  deselected:  false

[INFO]  Click  South Korea  clicksource:  null  type:  click  deselected:  true
```

iOS parity welcome!

[Testversion](http://migaweb.de/ti.map-android-4.5.0.zip)

### CODE REVIEW

**Comment by brentonhouse** [comment]
cc: @sgtcoolguy @garymathews

File: android/src/ti/map/TiUIMapView.java [hunk 1]
```diff
@@ -957,11 +954,12 @@ public void fireShapeClickEvent(LatLng clickPosition, IShape shapeProxy, String
 		}
 	}
 
-	public void fireClickEvent(Marker marker, AnnotationProxy annoProxy, String clickSource)
+	public void fireClickEvent(Marker marker, AnnotationProxy annoProxy, String clickSource, boolean deselected)
 	{
 		KrollDict d = new KrollDict();
 		String title = null;
 		String subtitle = null;
+		boolean isDeselected = (deselected) ? true : false;
```
Comment (garymathews) [inline android/src/ti/map/TiUIMapView.java]: Why redefine a `boolean`?

Comment (garymathews) [inline android/src/ti/map/TiUIMapView.java]: Why redefine a `boolean`?

Comment (m1ga) [reply android/src/ti/map/TiUIMapView.java]: :+1: you are right. Not needed

**Comment by sgtcoolguy** [comment]
Rebased on master, cleaned up commits some, bumped versions, attempted to add iOS parity - needs review by @vijaysingh-axway

File: apidoc/View.yml [hunk 2]
```diff
@@ -446,6 +446,13 @@ events:
             Longitude of the clicked annotation or the point clicked in the polygon, polyline and circle.
         type: Number
 
+      - name: deselected
+        summary: |
+            Will show if the annotation was selected (false) or deselected (true)
+        type: Boolean
+        since: "4.5.0"
+        platforms: [android]
```
Comment (vijaysingh-axway) [inline apidoc/View.yml]:
```suggestion
        platforms: [iphone, ipad, android]
```

**Comment by ssekhri** [comment]
FR passed. The deselected parameter, with value true or false, added to click event.
Verified on:
Mac OS: 10.14.5
SDK: 8.2.0.GA
Appc CLI: 7.1.1
JDK: 1.8.0_162
Node: 10.5.0
Studio: 5.1.4.201909061933
Xcode: 11.0
Device: iPhone X(13.0), iPhone 7Plus(12.3.1), Samsung S5(v6.0.1), Pixel 3(v10.0)emulator

**Pull request merged**